crates/sui-types/src/temporary_store.rs

```diff
@@ -958,8 +950,8 @@ impl<S: ObjectStore> TemporaryStore<S> {
     ) -> GasCostSummary {
         // at this point, we have done *all* charging for computation,
         // but have not yet set the storage rebate or storage gas units
-        assert!(gas_status.storage_rebate() == 0);
-        assert!(gas_status.storage_gas_units() == 0);
+        // assert!(gas_status.storage_rebate() == 0);
```

Comment:
not sure if we remove those or not. Those asserts were completely harmless and there more for documentation than anything else. At the time we introduced them this was only called after execution.
Later we have added a re-entrancy that is causing the assert to fire.
I am going to look in more details how to be more correct but wonder what people think about turning them off